### PR TITLE
Map blocks with functions with multiple outputs

### LINKED ISSUE #72: Create multiple output arrays with map_blocks
If a user provides a map function that produces multiple outputs, e.g.

```python
def f(block):
    return block - 1, block + 1
```

Then they should be able to specify that somehow in the call to mapblocks

```python
a, b = x.map_blocks(f, blockshape=[(10, 10), (10, 10)])
```

and receive two dask `Array` objects that share a common dask graph.

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #72

This map function produces many dask arrays from a block function that produces
many blocks.

Example
-------

    >>> d = da.ones(10, blockshape(5,))

    >>> def func(blk):
    ...     return (blk + 1, blk - 1)

    >>> a, b = da.map_blocks_multiple_outputs(d, func, nout=2)

### CODE REVIEW

File: dask/array/core.py [hunk 1]
```diff
@@ -361,6 +361,69 @@ def rec_concatenate(arrays, axis=0):
     return np.concatenate(arrays, axis=axis)
 
 
+def map_blocks_multiple_outputs(x, func, blockshapes=None, blockdimss=None,
+                                nout=None):
+    """ Map a multi-output function across all blocks of a dask array
+
+    Same as ``map_blocks`` but accepts a function that returns an iterable of
```
Comment (shoyer) [inline dask/array/core.py]: probably better to say "sequence" than "iterable"

**Comment by shoyer** [comment]
this looks potentially useful -- is something holding it up?

**Comment by mrocklin** [comment]
I think I was concerned that it didn't handle some non-standard cases.  We could merge it but keep it out of the `da` namespace.  Is this something you're likely to fool around with if it gets in?

**Comment by mrocklin** [comment]
Stale.  Closing.  Please reopen on renewed interest.

**Pull request closed without merging**